### PR TITLE
chore: xDD wrapper for the external API (#384)

### LINKED ISSUE #384: [TASK]: xDD wrapper for the external API
**Describe the task**
Create xDD wrapper for the external API. Be mindful this is a template for future external API.

**Task Items**
- [ ] list of items for completion
- [ ] another item

**Goal**
A clear and concise description of what the end result must have for it to be approved as complete.

**Screenshots**
If applicable, add screenshots or designs

**Additional context**
Add any other context about the task here

### PR DESCRIPTION
**Pull request by dvince2**
Adding the documents and extraction service proxy api, as well as the needed configurations to build and run the service

# Description

Adding a new service layer to proxy the xdd document and extraction service.  This PR also requires changes to be made to the orchestration project before it can be committed

Resolves #384

### CODE REVIEW

File: packages/services/document-server/build.gradle [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	implementation "io.quarkus:quarkus-resteasy-jsonb"
+	implementation "io.quarkus:quarkus-smallrye-openapi"
+	implementation "io.quarkus:quarkus-arc"
+
+	// REST client
+	implementation "io.quarkus:quarkus-rest-client"
+
+	// Auth dependencies
+	// implementation "io.quarkus:quarkus-oidc"
+	//implementation "io.quarkus:quarkus-keycloak-authorization"
```
Comment (YohannParis) [inline packages/services/document-server/build.gradle:25]: Why not removing it if not in use?

File: packages/services/document-server/build.gradle [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+java {
+	sourceCompatibility = JavaVersion.VERSION_17
+	targetCompatibility = JavaVersion.VERSION_17
+}
+
+test {
+	systemProperty "java.util.logging.manager", "org.jboss.logmanager.LogManager"
+}
+compileJava {
```
Comment (YohannParis) [inline packages/services/document-server/build.gradle:51]: Missing a space between those two sets of definition.

File: packages/services/document-server/docker/Dockerfile.jvm [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+# - GC_MAX_METASPACE_SIZE: The maximum metaspace size. (example: "100")
+# - GC_CONTAINER_OPTIONS: Specify Java GC to use. The value of this variable should
+#   contain the necessary JRE command-line options to specify the required GC, which
+#   will override the default of `-XX:+UseParallelGC` (example: -XX:+UseG1GC).
+# - HTTPS_PROXY: The location of the https proxy. (example: "myuser@127.0.0.1:8080")
+# - HTTP_PROXY: The location of the http proxy. (example: "myuser@127.0.0.1:8080")
+# - NO_PROXY: A comma separated lists of hosts, IP addresses or domains that can be
+#   accessed directly. (example: "foo.example.com,bar.example.com")
+#
+###
```
Comment (YohannParis) [inline packages/services/document-server/docker/Dockerfile.jvm:77]: Can we remove those comments? Or do they have any purpose other than being a Quarkus default?
@dgauldie?

File: packages/services/document-server/docker/Dockerfile.jvm [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'
+
+
+# We make four distinct layers so if there are application changes the library layers can be re-used
+COPY --chown=185 build/quarkus-app/lib/ /deployments/lib/
+COPY --chown=185 build/quarkus-app/*.jar /deployments/
+COPY --chown=185 build/quarkus-app/app/ /deployments/app/
+COPY --chown=185 build/quarkus-app/quarkus/ /deployments/quarkus/
+
+EXPOSE 3030
```
Comment (YohannParis) [inline packages/services/document-server/docker/Dockerfile.jvm:89]: Could we add those port to the main README.md of the repo?
Maybe a list of all Quarkus services and their ports, so we don't have to dig deep to find them.

File: packages/services/document-server/docker/Dockerfile.native [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+#
+# Then, build the image with:
+#
+# docker build -f src/main/docker/Dockerfile.native -t quarkus/document-server .
+#
+# Then run the container using:
+#
+# docker run -i --rm -p 8080:8080 quarkus/document-server
+#
+###
```
Comment (YohannParis) [inline packages/services/document-server/docker/Dockerfile.native:16]: Again this is all handle by the bake file, we don't really need them here.

File: packages/services/document-server/src/main/java/software/uncharted/terarium/documentserver/proxies/xdd/DocumentProxy.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
+package software.uncharted.terarium.documentserver.proxies.xdd;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@RegisterRestClient(configKey = "xdd-document-service")
```
Comment (YohannParis) [inline packages/services/document-server/src/main/java/software/uncharted/terarium/documentserver/proxies/xdd/DocumentProxy.java:9]: Is this where you make reference to the xDD API URL in the `resources/application.properties`?
Then _Microprofile_ automaticaly does the REST call with the `@GET` annotation?

**Review by YohannParis — APPROVED** [review]
LGTM

File: packages/services/document-server/build.gradle (hunk 1 above)
Comment (dvince2) [reply packages/services/document-server/build.gradle]: Good point! Thanks

**Review by YohannParis — APPROVED** [review]

**Review by chris-dickson — CHANGES_REQUESTED** [review]

File: packages/services/document-server/src/main/java/software/uncharted/terarium/documentserver/models/xdd/XDDResponseOK.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+import java.io.Serializable;
+
+@Data
+@Accessors(chain = true)
+public class XDDResponseOK implements Serializable {
+	private Number v;
```
[the remark this answers is not in the document]
Comment (dvince2) [reply packages/services/document-server/src/main/java/software/uncharted/terarium/documentserver/models/xdd/XDDResponseOK.java:11]: These were copied from the original document code from HMI-Server.  Following up with what this is now to add documentation on it

[the remark this answers is not in the document]
Comment (dvince2) [reply packages/services/document-server/src/main/java/software/uncharted/terarium/documentserver/models/xdd/XDDResponseOK.java:14]: Confirmed version

**Review by chris-dickson — DISMISSED** [review]

**Review by chris-dickson — APPROVED** [review]

**Pull request merged**